### ISSUE TITLE
[Android] Wrong arraybuffer is generated

### ISSUE THREAD

**Issue by mlecoq** [mrousavy/react-native-blob-jsi-helper#4]
Hi,

I am using pdf-lib (https://github.com/Hopding/pdf-lib) to manage pdf on my rn app

pdf-lib takes an arraybuffer as param, for the moment I convert my local pdf files as base64 and then to arraybuffer (using https://github.com/beatgammit/base64-js) to use them as params in pdf-lib methods

I wanted to try your lib to check if I could have better perfs. Unfortunately generated arraybuffer with `react-native-blob-jsi-helper` cannot be used with pdf-lib. I have made a small project to check the diff between using `react-native-blob-jsi-helper` and using base64 : https://github.com/mlecoq/jsi-blob-issue

When I log the generated buffer with `react-native-blob-jsi-helper`, first values seem correct but then I only have zeros:

```
[37, 80, 68, 70, 45, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ....
```

compared to correct arraybuffer 

```
[37, 80, 68, 70, 45, 49, 46, 51, 13, 10, 37, 226, 227, 207, 211, 13, 10, 13, 10, 49, 32, 48, 32, 111, 98, 106, 13, 10, 60, 60, 13, 10, 47, 84, 121, 112, 101, 32, 47, 67, 97, 116, 97, 108, 111, 103, 13, 10, 47, 79, 117, 116, 108, 105, 110, 101, 115, 32, 50, 32, 48, 32, 82, 13, 10, 47, 80, 97, 103, 101, 115, 32, 51, 32, 48, 32, 82, 13, 10, 62, 62, 13, 10, 101, 110, 100, 111, 98, 106, 13, 10, 13, 10, 50, 32, 48, 32, 111, 98, 106, 13, 10, 60, 60, 13, 10, 47, 84, 121, 112, 101, 32, 47, 79, 117, 116, 108, 105, 110, 101, 115, 13, 10, 47, 67, 111, 117, 110, 116, 32, 48, 13, 10, 62, 62, 13, 10, 101, 110, 100, 111, 98, 106, 13, 10, 13, 10, 51, 32, 48, 32, 111, 98, 106, 13, 10, 60, 60, 13, 10, 47, 84, 121, 112, 101, 32, 47, 80, 97, 103, 101, 115, 13, 10, 47, 67, ...
```

**Comment by mrousavy** [comment]
hey! Is this an ArrayBufferView?

**Comment by mlecoq** [comment]
Yes they are both ArrayBufferView (Uint8Array) - `pdf-lib` was not able to parse generated buffer with `react-native-blob-jsi-helper` at index 5 (it's why I checked the content)

**Comment by mlecoq** [comment]
fyi this issue is specific to `android` - on iOS I have the same content

**Comment by mlecoq** [comment]
I am not fluent in cpp but I have remarked that in https://github.com/mrousavy/react-native-blob-jsi-helper/blob/master/android/src/main/cpp/cpp-adapter.cpp 

the following statement

`memcpy(arrayBuffer.data(runtime), reinterpret_cast<byte*>(bytes), 5);`

5 is hardcoded and I have 5 good values

**Comment by mlecoq** [comment]
That is the error, I will make a PR

**Comment by mrousavy** [comment]
ahahahah such a dumb mistake I'm sorry man

**Comment by kesha-antonov** [comment]
Hi @mrousavy 
Please release the fix

**Comment by bfricka** [comment]
@mrousavy ? Due to the way this project is structured (scripts), using git commit isn't enough. The only way to access this fix is to fork.

**Comment by kesha-antonov** [comment]
> @mrousavy ? Due to the way this project is structured (scripts), using git commit isn't enough. The only way to access this fix is to fork.

```
"react-native-blob-jsi-helper": "mrousavy/react-native-blob-jsi-helper#9146ec59cfeae347b22c51512204c94c45b084c1",

import { getArrayBufferForBlob } from 'react-native-blob-jsi-helper'
```

This works for me for now

**Comment by mrousavy** [comment]
Released 0.2.1 to npm 🎉
